packages/core/core/src/loaders/apis.ts

```diff
@@ -168,21 +172,22 @@ const loadDir = async (dir: string) => {
 
     const key = basename(fd.name, extname(fd.name));
 
-    root[normalizeName(key)] = await loadFile(join(dir, fd.name));
+    root[normalizeName(key)] = (await loadFile(join(dir, fd.name))).result;
   }
 
-  return root;
+  return { result: root };
 };
 
-const loadFile = (file: string) => {
+// because this is async and its contents are dynamic, we must return it as an array to avoid a property called `then` being interpreted as a Promise
+const loadFile = async (file: string): Promise<{ result: unknown }> => {
```

Comment:
by making the function async aren't you removing the need for that result 🤔  if you await loadFile no it won't be call then on any of the imported code because it's wrapped I think

Reply:
Yeah, I got the whole problem wrong because it seemed that something weird within importDefault was making it async sometimes, but that wasn't actually the case.

Reply:
Ok, nevermind, I checked.  I did have a useless await on `importDefault` but `readJSON` is async, that's why I put it explicitly.

So everything else needs to be here. Including this `result`, because `await Promise.all` waits for all the nested promises to settle, meaning it tries to call the `.then()` from loadFIles, even if it were made sync with `readJSONSync`. So we have to add this layer where there is no function called 'then' or it will get called

Reply:
Arf ok I thought we could avoid it 🤷

Reply:
So the big question is, is allowing this worth a risk that someone could make a similar bug in the future? I could add an api test with a content type called "then" but it almost feel a bit silly to work on this considering it's just one keyword we can ban. I actually did this "as a community member" on the weekend for that reason 😆

Reply:
Ok, I'll leave `then` restricted (for now) and merge the actual fix, with a note that it's only restricted because it's a keyword but no longer an issue